src/Microsoft.Health.Fhir.Liquid.Converter/Models/PartialDateTime.cs

```diff
@@ -37,7 +37,8 @@ public PartialDateTime(string input, DateTimeType type = DateTimeType.Fhir)
             int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
             int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
             int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
-            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value) : 0;
+            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value.Length > 3 ? groups["millisecond"].Value.Substring(0, 3) : groups["millisecond"].Value) : 0;
```

Comment:
The 4th digit is invalid?

Reply:
4th is invalid when new a datetime object in .net.
So we both save string part of millisecond and 3 digits as int value here.